wallet.js

```diff
@@ -0,0 +1,270 @@
+ * - state: When the state of the Wallet changes.
+ * - new-tx: When a new tx arrives.
+ * - update-tx: When a known tx is updated. Usually, it happens when one of its outputs is spent.
+ **/
+class Wallet extends EventEmitter {
+  constructor({ server, seed }) {
+    super();
+
+    this.state = this.CLOSED;
+
+    this.onConnectionChange = this.onConnectionChange.bind(this);
+    this.handleWebsocketMsg = this.handleWebsocketMsg.bind(this);
+    this.onAddressesLoaded = this.onAddressesLoaded.bind(this);
+
+    this.server = server;
+    this.seed = seed;
+
+    this.passphrase = '';
+    this.pinCode = '123';
+    this.password = '123';
+  }
+
+  /**
+   * Called when loading the history of transactions.
+   * It is called every HTTP Request to get the history of a set of addresses.
+   * Usually, this is called multiple times.
+   * The `historyTransactions` contains all transactions, including the ones from a previous call to this method.
+   *
+   * @param {Object} result {historyTransactions, allTokens, newSharedAddress, newSharedIndex, addressesFound}
+   **/
+  onAddressesLoaded(result) {
+    //console.log('result', result);
+  }
+
+  /**
+   * Called when the connection to the websocket changes.
+   * It is also called if the network is down.
+   **/
+  onConnectionChange(value) {
+    if (value) {
+      this.setState(Wallet.SYNCING);
+      hathorLib.wallet.loadAddressHistory(0, hathorLib.constants.GAP_LIMIT).then(() => {
+        this.setState(Wallet.READY);
+      }).catch((error) => {
+        throw error;
+      })
+    }
+  }
+
+  getAddressToUse() {
+    return hathorLib.wallet.getAddressToUse();
+  }
+
+  getCurrentAddress() {
+    return hathorLib.wallet.getCurrentAddress();
+  }
+
+  /**
+   * Called when a new message arrives from websocket.
+   **/
+  handleWebsocketMsg(wsData) {
+    if (wsData.type === 'wallet:address_history') {
+      this.onNewTx(wsData);
+    }
+  }
+
+  reloadData() {
+    console.log('reloadData');
+  }
+
+  getAllBalances() {
+    return hathorLib.storage.getItem('local:tokens:balance');
+  }
+
+  getBalance(tokenUid) {
+    return this.getAllBalances[tokenUid];
+  }
+
+  setState(state) {
+    this.state = state;
+    this.emit('state', state);
+  }
+
+  onNewTx(wsData) {
+    const newTx = wsData.history;
+
+    // TODO we also have to update some wallet lib data? Lib should do it by itself
+    const walletData = hathorLib.wallet.getWalletData();
+    const historyTransactions = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+    const allTokens = 'allTokens' in walletData ? walletData.allTokens : [];
+
+    let isNewTx = false;
+    if (newTx.tx_id in historyTransactions) {
+      isNewTx = true;
+    }
+
+    hathorLib.wallet.updateHistoryData(historyTransactions, allTokens, [newTx], null, walletData);
+
+    if (isNewTx) {
+      this.emit('new-tx', newTx);
+    } else {
+      this.emit('update-tx', newTx);
+    }
+    return;
```

Comment:
return here? What about the rest of the code?

Reply:
Those are incomplete pieces of code. I put the remaining code as comments.